dfget/core/uploader/uploader.go

```diff
@@ -286,6 +287,37 @@ func deleteExpiredFile(api api.SupernodeAPI, path string, info os.FileInfo,
 	return false
 }
 
+func captureQuitSignal(cfg *config.Config) {
+	c := make(chan os.Signal)
+	signal.Notify(c, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGSTOP)
+	s := <-c
+	logrus.Infof("capture stop signal: %s, will shutdown...", s)
+
+	// tell supernode this peer node is down and delete related files.
+	p2p.syncTaskMap.Range(func(key, value interface{}) bool {
+		task, ok := value.(*taskConfig)
+		if ok {
+			p2p.api.ServiceDown(task.superNode, task.taskID, task.cid)
+			serviceFile := helper.GetServiceFile(key.(string), task.dataDir)
+			os.Remove(serviceFile)
+			logrus.Infof("shutdown, remove task id:%s file:%s",
+				task.taskID, serviceFile)
+		}
+		return true
+	})
+
+	shutdown(cfg)
+}
+
+func shutdown(cfg *config.Config) {
```

Comment:
If we only use one parameter of `config.Config.RV.MetaPath` in all implementation of `shutdown` , I do not think it is a good way to input the whole `config.Config` instance. How about just only inputting the essential and sufficient parameters? I will make the code much more accurate and much more readable.

Reply:
agree.